hatchet_sdk/context.py

```diff
@@ -198,17 +200,40 @@ def spawn_workflow(
 
         return self.admin_client.run_workflow(workflow_name, input, trigger_options)
 
-    def _log(self, line: str):
+    def _log(self, line: str) -> (bool, Exception):  # type: ignore
         try:
             self.event_client.log(message=line, step_run_id=self.stepRunId)
+            return True, None
         except Exception as e:
-            logger.error(f"Error logging: {e}")
+            # we don't want to raise an exception here, as it will kill the log thread
+            return False, e
 
-    def log(self, line: str):
+    def log(self, line, raise_on_error: bool = False):
         if self.stepRunId == "":
             return
 
-        self.logger_thread_pool.submit(self._log, line)
+        if not isinstance(line, str):
+            line = json.dumps(line)
+
+        future: Future = self.logger_thread_pool.submit(self._log, line)
```

Comment:
Shouldn't this future be a multithreading `Future`, not `asyncio.Future`?

Reply:
good catch, vs code seems to love asyncio for imports... gotta be careful